kernel/arch/dreamcast/hardware/biosfont.c

```diff
@@ -70,20 +75,25 @@ int bfont_set_32bit_mode(int on) {
     return rv;
 }
 
-/* A little assembly that grabs the font address */
-extern uint8* get_font_address(void);
-__asm__("	.text\n"
-        "	.align 2\n"
-	".globl _get_font_address\n"
-        "_get_font_address:\n"
-        "	mov.l	syscall_b4,r0\n"
-        "	mov.l	@r0,r0\n"
-        "	jmp	@r0\n"
-        "	mov	#0,r1\n"
-        "\n"
-        "	.align 4\n"
-        "syscall_b4:\n"
-        "	.long	0x8c0000b4\n");
+mutex_t _g1_bfont_mutex = MUTEX_INITIALIZER;
+
+int lock_bfont(void) {
+    if(mutex_lock(&_g1_bfont_mutex) == -1) return -1;
```

Comment:
I think better replace this mutex with shared g1_ata_mutex that used in CD-ROM and G1ATA drivers.
Because this double locking does not affect the G1ATA driver in any way. But you need to block the G1ATA driver too, since it uses the same bus.

Reply:
Do you think it just needs to be renamed _g1_mutex since it seems like it will be used for all things connected to the G1 Bus?

Reply:
Yes, this makes sense, since it is not only for ATA.

Reply:
I think that rename should be in a future PR where we probably look at the flashrom.c and cdrom.c files for GDROM dma work.

Reply:
> Do you think it just needs to be renamed _g1_bus_mutex since it seems like it will be used for all things connected to the G1 Bus (biosfont, flashrom, cd, ata, etc)?

Yes, that's right. I'd made it explicitly to demonstrate the mutex for the font, but it should be a shared one for all g1 devices. It likely makes the most sense for it to live in the same syscalls.c file as it's basically an interface into a set of syscallls. It would only be called by things that are already using syscalls anyways so that would streamline things too.